docs/datatypes.md

```diff
@@ -0,0 +1,253 @@
+
+Defining New Types
+==================
+
+Unlike most Lisps, Irken's type system is based on 'algebraic
+datatypes', where complex types are built from simpler types.
+
+New types are created with the `datatype` special form:
+
+    (datatype year
+      (:known int)
+      (:approx int)
+      (:range int int)
+      (:unknown)
+      )
+
+Here, we've created a new datatype `year`, which has four
+different 'variants'.  To create a 'known' year:
+
+    (year:known 1972)
```

Comment:
It might help to say that this whole _expression_ is of type year. Or "type year, variant  known" (if that's accurate). To the very ignorant (me) I want to read that as `year:known` getting _assigned_ the value of `1972`.